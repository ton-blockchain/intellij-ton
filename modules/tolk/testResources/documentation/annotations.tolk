/// Function with inline annotation
@inline
fun inlineFunction(): int {
    return 42;
}

/// Function with inline_ref annotation
@inline_ref
fun inlineRefFunction(): int {
    return 24;
}

/// Function with noinline annotation
@noinline
fun noinlineFunction(): int {
    return 100;
}

/// Function with pure annotation
@pure
fun pureFunction(x: int): int {
    return x * 2;
}

/// Deprecated function
@deprecated("Use newFunction() instead")
fun oldFunction(): void {
}

/// Function with overflow policy
@overflow1023_policy("suppress")
fun functionWithOverflowPolicy(): void {
}

/// Function with bounced message policy
@on_bounced_policy("manual")
fun functionWithBouncedPolicy(): void {
}

/// Function with method ID
@method_id(123)
fun getBalance(): int {
    return 0;
}

/// Function with multiple annotations
@method_id(456)
@inline
@pure
fun multiAnnotatedFunction(): int {
    return 1;
}

/// Constant with deprecated annotation
@deprecated("Use NEW_CONST instead")
const OLD_CONST = 42;
