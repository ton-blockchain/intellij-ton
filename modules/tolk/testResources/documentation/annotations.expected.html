<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Tolk Documentation Test Results</title>
    <style>
        body { font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif; margin: 20px; background: #fafafa; }
        h1 { color: #2c3e50; border-bottom: 3px solid #3498db; padding-bottom: 10px; }
        .element { margin-bottom: 25px; border: 1px solid #ddd; padding: 20px; background: white; border-radius: 8px; box-shadow: 0 2px 4px rgba(0,0,0,0.1); }
        .element-title { font-weight: bold; color: #2c3e50; margin-bottom: 15px; font-size: 14px; background: #ecf0f1; padding: 8px; border-radius: 4px; }
        .documentation { background: #f8f9fa; padding: 15px; border-radius: 6px; border-left: 4px solid #3498db; font-family: 'JetBrains Mono', 'Fira Code', monospace; line-height: 1.6; }
        
        /* Tolk syntax highlighting */
        .keyword { color: #d73a49; font-weight: bold; }
        .primitive { color: #005cc5; font-weight: 500; }
        .struct { color: #22863a; font-weight: 500; }
        .function { color: #6f42c1; font-weight: 500; }
        .parameter { color: #e36209; }
        .type-alias { color: #d73a49; }
        .type-parameter { color: #6f42c1; font-style: italic; }
        .annotation { color: #6a737d; }
        .string { color: #032f62; }
        .number { color: #005cc5; }
        .comment { color: #6a737d; font-style: italic; }
        .constant { color: #005cc5; font-weight: 500; }
        .global-variable { color: #e36209; font-weight: 500; }
        .identifier { color: #24292e; }
        
        /* Documentation markup */
        .definition-start { border-top: 2px solid #e1e4e8; margin-top: 10px; padding-top: 10px; }
        .definition-end { border-bottom: 1px solid #e1e4e8; margin-bottom: 10px; padding-bottom: 10px; }
        .content-start { margin-top: 15px; }
    </style>
</head>
<body>
    <h1>Tolk Documentation Test Results</h1>
    <div class="element">
        <div class="element-title">Annotation: inline</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">inline</span></pre></div><div class='content'>Function with this annotation will be automatically inlined during compilation</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: inlineFunction</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">inline</span>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">inlineFunction</span><span style="">()</span>: <span style="color:#000080;font-weight:bold;">int</span></pre></div><div class='content'><p>Function with inline annotation</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: inline_ref</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">inline_ref</span></pre></div><div class='content'>Function with this annotation will be automatically inlined by reference during compilation</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: inlineRefFunction</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">inline_ref</span>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">inlineRefFunction</span><span style="">()</span>: <span style="color:#000080;font-weight:bold;">int</span></pre></div><div class='content'><p>Function with inline_ref annotation</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: noinline</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">noinline</span></pre></div><div class='content'>Function with this annotation will not be inlined even if compiler can inline it</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: noinlineFunction</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">noinline</span>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">noinlineFunction</span><span style="">()</span>: <span style="color:#000080;font-weight:bold;">int</span></pre></div><div class='content'><p>Function with noinline annotation</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: pure</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">pure</span></pre></div><div class='content'>Function with this annotation has no side effects and can be optimized away by the compiler</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Parameter: x</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#000000;">x</span>: <span style="color:#000080;font-weight:bold;">int</span></pre></div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: pureFunction</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">pure</span>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">pureFunction</span><span style="">(</span><span style="color:#000000;">x</span>: <span style="color:#000080;font-weight:bold;">int</span><span style="">)</span>: <span style="color:#000080;font-weight:bold;">int</span></pre></div><div class='content'><p>Function with pure annotation</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: deprecated</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">deprecated</span><span style="">(</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="color:#008000;font-weight:bold;">Use newFunction() instead</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="">)</span></pre></div><div class='content'>Symbol with this annotation is deprecated and should not be used in new code. First string argument is a reason for deprecation as a string literal.</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: void</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">void</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: oldFunction</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">deprecated</span><span style="">(</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="color:#008000;font-weight:bold;">Use newFunction() instead</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="">)</span>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">oldFunction</span><span style="">()</span>: <span style="color:#000080;font-weight:bold;">void</span></pre></div><div class='content'><p>Deprecated function</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: overflow1023_policy</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">overflow1023_policy</span><span style="">(</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="color:#008000;font-weight:bold;">suppress</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="">)</span></pre></div><div class='content'>Defines the policy for handling potential builder overflow. Right now, only "suppress" value is supported. See <a href="https://docs.ton.org/v3/documentation/smart-contracts/tolk/tolk-vs-func/pack-to-from-cells#what-if-data-exceeds-1023-bits">documentation</a> for more details</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: void</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">void</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: functionWithOverflowPolicy</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">overflow1023_policy</span><span style="">(</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="color:#008000;font-weight:bold;">suppress</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="">)</span>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">functionWithOverflowPolicy</span><span style="">()</span>: <span style="color:#000080;font-weight:bold;">void</span></pre></div><div class='content'><p>Function with overflow policy</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: on_bounced_policy</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">on_bounced_policy</span><span style="">(</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="color:#008000;font-weight:bold;">manual</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="">)</span></pre></div><div class='content'>Defines the policy for handling bounced messages. Right now, only "manual" value is supported.</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: void</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">void</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: functionWithBouncedPolicy</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">on_bounced_policy</span><span style="">(</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="color:#008000;font-weight:bold;">manual</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="">)</span>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">functionWithBouncedPolicy</span><span style="">()</span>: <span style="color:#000080;font-weight:bold;">void</span></pre></div><div class='content'><p>Function with bounced message policy</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: method_id</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">method_id</span><span style="">(</span><span style="color:#0000ff;">123</span><span style="">)</span></pre></div><div class='content'>Specifies the method ID (as a number literal) for the function in smart contract interface. See <a href="https://docs.ton.org/v3/guidelines/smart-contracts/get-methods">documentation</a> for more details</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: getBalance</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">method_id</span><span style="">(</span><span style="color:#0000ff;">123</span><span style="">)</span>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">getBalance</span><span style="">()</span>: <span style="color:#000080;font-weight:bold;">int</span></pre></div><div class='content'><p>Function with method ID</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: method_id</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">method_id</span><span style="">(</span><span style="color:#0000ff;">456</span><span style="">)</span></pre></div><div class='content'>Specifies the method ID (as a number literal) for the function in smart contract interface. See <a href="https://docs.ton.org/v3/guidelines/smart-contracts/get-methods">documentation</a> for more details</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: inline</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">inline</span></pre></div><div class='content'>Function with this annotation will be automatically inlined during compilation</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: pure</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">pure</span></pre></div><div class='content'>Function with this annotation has no side effects and can be optimized away by the compiler</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: multiAnnotatedFunction</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">method_id</span><span style="">(</span><span style="color:#0000ff;">456</span><span style="">)</span>
<span style="color:#808000;">@</span><span style="color:#808000;">inline</span>
<span style="color:#808000;">@</span><span style="color:#808000;">pure</span>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">multiAnnotatedFunction</span><span style="">()</span>: <span style="color:#000080;font-weight:bold;">int</span></pre></div><div class='content'><p>Function with multiple annotations</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Annotation: deprecated</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">deprecated</span><span style="">(</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="color:#008000;font-weight:bold;">Use NEW_CONST instead</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="">)</span></pre></div><div class='content'>Symbol with this annotation is deprecated and should not be used in new code. First string argument is a reason for deprecation as a string literal.</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Const: OLD_CONST</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#808000;">@</span><span style="color:#808000;">deprecated</span><span style="">(</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="color:#008000;font-weight:bold;">Use NEW_CONST instead</span><span style="color:#008000;font-weight:bold;">&quot;</span><span style="">)</span>
<span style="color:#000080;font-weight:bold;">const</span> <span style="color:#660e7a;">OLD_CONST</span>: <span style="color:#000080;font-weight:bold;">int</span> = <span style="color:#0000ff;">42</span></pre></div><div class='content'><p>Constant with deprecated annotation</p>
</div>
        </div>
    </div>
</body>
</html>
